src/app/ui/nav/Nav.tsx

```diff
@@ -18,12 +19,47 @@ const NAV_MENU = [
   },
 ];
 
+const NAV_ITEM_HEIGHT = 25;
+
 export default function Nav() {
+  const currentPagePath = window.location.pathname;
+
+  let activeIndex = 0;
```

Comment:
here is a good case for `useMemo` - a hook used for memoizing variables

in `useMemo`, u write a function that returns the value u want. and in the square brackets, you put variables that it depends on. the hook will only update if any variables in the brackets change.

```typescript
const activeIndex = useMemo(() => {
  let index = 0;

  NAV_MENU.forEach((item, i) => {
    if (currentPagePath === item.link) {
      index = i;
      return; // small optimization - break out of loop once u find a match
    }
  });

  return index;
}, [currentPagePath]);
```

Reply:
i dont believe in usememo

Reply:
i dont believe in memoization ... smh

Reply:
thank u cs god